File: aziz/python/ch04/reverse-number.py

def reverseNumber(n):
    neg = False
    if n < 0:
        neg = True
        n = - n

    ans = 0

    while n > 0:
        ans = ans * 10 + (n % 10)
        n = n / 10

    return ans if not neg else - ans

print reverseNumber(123450)

File: aziz/php/ch04/reverse-number.php
<?php

function reverseNumber($n) {
    $isNegative = $n < 0;

    $result = 0;
    $remaining = abs($n);

    while($remaining) {
        $result = $result * 10 + $remaining % 10;
        $remaining = floor($remaining/ 10);
    }

    return $isNegative ? - $result : $result;
}


echo reverseNumber(1234);